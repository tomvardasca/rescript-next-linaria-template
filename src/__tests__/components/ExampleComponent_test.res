open Jest
open Expect
open ReactTestingLibrary

test("Component matches snapshot", () => {
  
  <ExampleComponent /> |> render |> container |> expect |> toMatchSnapshot
})

test("Component renders hook", () => {
  open JestDom

  let result = <ExampleComponent /> |> render

  result |> getByText(~matcher=#RegExp(%re("/Count: 0/"))) |> expect |> toBeInTheDocument
})

test("Component renders hook after click", () => {
  open JestDom
  let result = <ExampleComponent /> |> render

  act(() => result |> getByText(~matcher=#RegExp(%re("/Click me/"))) |> FireEvent.click |> ignore)

  result |> getByText(~matcher=#RegExp(%re("/Count: 1/"))) |> expect |> toBeInTheDocument
})
