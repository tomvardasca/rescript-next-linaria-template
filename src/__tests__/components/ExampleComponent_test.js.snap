// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component matches snapshot 1`] = `
<div>
  <div>
    <button
      class="button_0"
    >
      Click me
    </button>
    Count: 0
  </div>
</div>
`;
